files/usr/share/jeos-firstboot/raspberrywifi

```diff
@@ -122,23 +120,40 @@ WIRELESS_ESSID='$wlan_network'
 WIRELESS_MODE='Managed'
 EOF
 
-	if [ $wlan_auth_mode = "WPA-PSK" ]; then
-		echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
-	fi
+		if [ $wlan_auth_mode = "WPA-PSK" ]; then
+			echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
+		fi
 
-	if [ $wlan_auth_mode = "WPA-EAP" ]; then
-		echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
-		echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
-		echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
-	fi
+		if [ $wlan_auth_mode = "WPA-EAP" ]; then
+			echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
+			echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
+			echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
+		fi
 
-	run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		d --infobox $"Connecting to wireless network ..." 3 38 || true
+
+		run ifdown $wlan_device &>/dev/null || true
+		# Disabling exit on error to capture exit code
+		set +e
```

Comment:
Bash 101:

```
if ifup $wlan_device &>/dev/null; then
...
fi
```

Reply:
again, it's broke down the way it is on purpose, to make it clearer,  pure stylistic choice.

Reply:
No, this is simply bad code.